app/src/main/java/com/guap/vkr/playlistmaker/search/data/dto/ResponseStatus.kt

```diff
@@ -0,0 +1,6 @@
+package com.guap.vkr.playlistmaker.search.data.dto
+
+sealed class ResponseStatus<T>(val data: T? = null, val hasError: Boolean? = false) {
+    class Success<T>(data: T): ResponseStatus<T>(data)
+    class Error<T>(hasError: Boolean?, data: T? = null): ResponseStatus<T>(data, hasError)
```

Comment:
🍏 Уже сам статус Error говорить о том, что произошла ошибка:
```suggestion
    class Error<T>(data: T? = null): ResponseStatus<T>(data, true)
```